te_chef.py

```diff
@@ -35,50 +62,228 @@ def construct_channel(self, **kwargs):
+    content_items = doc.select(".post_title_sub .current_post")
+    slugs_added = set()
+
+    for content in content_items:
+        slug = content.select_one(".get_post_title")["value"]
+
+        # Skip duplicates ... seems like the Touchable Earth website has them!
+        if slug in slugs_added:
+            continue
         else:
-            child_node = nodes.VideoNode(
-                source_id=child_source_node["id"],
-                title=child_source_node["title"],
-                license=child_source_node["license"],
-                description=create_description(child_source_node),
-                derive_thumbnail=True,
-                files=[files.YouTubeVideoFile(youtube_id=child_source_node['youtube_id'])],
-            )
-            for language in child_source_node['subtitle_langs']:
-                child_node.add_file(files.YouTubeSubtitleFile(youtube_id=child_source_node['youtube_id'], language=language))
-            parent.add_child(child_node)
-
-    return parent
+            slugs_added.add(slug)
+
+        title = content.select_one(".get_post_title2")["value"]
+        site_url = content.select_one(".site_url")["value"]
+        url = "%s/%s" % (site_url, slug)
+
+        content_node = scrape_content(title, url)
+        if content_node:
+            category_node.add_child(content_node)
+
+    return category_node
+
+
+def scrape_content(title, content_url):
+    """
+    title: Boys' clothing
+    content_url: http://www.touchableearth.org/china-culture-boys-clothing/
+    """
+    print("    Scraping content node: %s (%s)" % (title, content_url))
+
+    doc = get_parsed_html_from_url(content_url)
+    if not doc:  # 404
+        return None
+
+    description = create_description(doc.select_one(".tab-content"))
+    base_node_attributes = {
+        "source_id": content_url,
+        "title": title,
+        "license": TE_LICENSE,
+        "description": description,
+    }
+
+    youtube_iframe = doc.select_one(".video-container iframe")
+    if youtube_iframe:
+        youtube_url = doc.select_one(".video-container iframe")["src"]
+        youtube_id = get_youtube_id_from_url(youtube_url)
+
+        video_node = nodes.VideoNode(
+            **base_node_attributes,
+            derive_thumbnail=True,
+            files=[files.YouTubeVideoFile(youtube_id=youtube_id)],
+        )
+
+        # TODO(davidhu): Get subtitles in other languages
+
+        return video_node
+
+    img = doc.select_one(".uncode-single-media-wrapper img")
+    if img:
+        img_src = img["data-guid"] or img["src"]
+        destination = tempfile.mkdtemp()
+        download_file(img_src, destination, request_fn=make_request, filename="image.jpg")
+
+        with open(os.path.join(destination, "index.html"), "w") as f:
+            f.write("""
+                <html>
+                <head></head>
+                <body>
+                    <img src="image.jpg" style="width: 100%;" />
```

Comment:
By a `max-width` I'm guessing you want a fixed pixel max width (as opposed to `max-width: 100%`)? I made a rough stab with `1200px`. I'm not sure what the common screen sizes of Kolibri users would be ...